Chore: switch to google's uuid module

mikelolasagasti opened the pull request:
### 🤔 What's changed?

Project was using non-modularized version of `github.com/gofrs/uuid`. This change switches to module `github.com/google/uuid`.

`gofrs` has a `v5` version that could be used also, but I tend to use Google's module in my projects.

When using Google's module, `New()` is equal of doing `uuid.Must(uuid.NewRandom())`, so the change mimics the old `uuid.Must(uuid.NewV4())` code but it's shorter.

<!-- Describe your changes in detail -->

### ⚡️ What's your motivation? 

<!-- 
What motivated you to propose this change? Does it fix a bug? Add a new feature?
If it fixes an open issue, you can link to the issue here, e.g. "Fixes #99"
-->

Using updated modules in the project. 

### 🏷️ What kind of change is this?

<!--- Delete any options that are not relevant -->

- :bank: Refactoring/debt/DX (improvement to code design, tooling, etc. without changing behaviour)

### 📋 Checklist:

<!--- 
This is to help you remember all the little things we often forget to do!

Feel free to delete any tasks that are not relevant, or add new ones.
-->

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://github.com/cucumber/.github/tree/main?tab=coc-ov-file)

----

mpkorstanje reviewed (commented):
Looks good. But please also update the `CHANGELOG` with an entry under changed.

mikelolasagasti commented:
Updated `CHANGELOG` file, let me know if I need to change something else.

In the last version I also changed the import order in `messages_test.go` to make `gofumpt` pass.

mpkorstanje commented:
Cheers! 

If you are using Godog, you might want to update the version of messages used in Gherkin too. It's currently behind a few versions.
